### PR TITLE
Improved support for multiple currencies

### LINKED ISSUE #50: App crashes when switching from custom to default currency settings
In the Account Settings, unchecking "Use default currency settings" and entering custom currency works fine. Changing back to default by checking the box again crashes the app, if the currency is set to anything but `$`.

### PR DESCRIPTION
**Pull request by dospuntos**
This PR fixes some issues where custom currency settings where lost when closing the app. In addition, QuickTracker will show the total balance for each of the currencies, instead of adding them all.

- Custom currency settings are correctly saved
- Fixes #50 
- QuickTracker lists the balance per currency, not all combined
- Replaced custom window centering with CenterIn()

I'm not sure what the best way to list the QuickTracker information is, I opted for "Balance" followed by the currency symbol, but maybe there's another way that is more intuitive?

capitalBe

### CODE REVIEW

File: src/Database.cpp [hunk 1]
```diff
@@ -9,7 +9,8 @@
 #include "Import.h"
 #include "ScheduledTransData.h"
 #include "TimeSupport.h"
-#include "Transaction.h"
+// #include "Transaction.h"
```
[the remark this answers is not in the document]
Comment (dospuntos) [reply src/Database.cpp]: Sorry, I forgot it. Removed.

**Comment by humdingerb** [comment]
The defaults could be grabbed from the currently set locale setting. In fact, for the date format it may be sufficient to always use the locale setting and the remove the option completely.

Found an issue:
I created a new account with € as currency.
I entered a transaction with an amount of "1000". After tab to move to the next text entry, the Amount text field showed "$1000".  The transaction list shows correctly "1000€", but when double-clicking it, it's again "$1000" in the edit window.

**Comment by dospuntos** [comment]
> The defaults could be grabbed from the currently set locale setting. In fact, for the date format it may be sufficient to always use the locale setting and the remove the option completely.

Using system locale would be nice, and I hope to be able to implement it soon. Currently the default locale settings are 
hardcoded in several places and saved in the database, so it might require quite a lot of work. Good point about the date too, but there's a lot of code around converting between different date formats, so it might be better with a separate PR to change that.

> Found an issue: I created a new account with € as currency. I entered a transaction with an amount of "1000". After tab to move to the next text entry, the Amount text field showed "$1000". The transaction list shows correctly "1000€", but when double-clicking it, it's again "$1000" in the edit window.

Thanks, it turns out that the global variable `gCurrentLocale` wasn't updated when changing accounts. This should be fixed now.

**Comment by dospuntos** [comment]
Thanks @humdingerb , not sure if I got the rebase right, but feel free to merge if it's OK.

**Comment by korli** [comment]
@humdingerb you can choose "squash and merge" if you like.

**Comment by humdingerb** [comment]
@dospuntos : almost, you'd have to reorder the old last commit after the first before squashing those two. Now you squashed the third with the second commit... No matter though, we merge as is. :)

@korli : squash & merge is ofc possible, but I think it's normally preferred to keep the separate commits in a PR if possible.

**Pull request merged**